api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java

```diff
@@ -0,0 +1,219 @@
+        cd.setQuery(sqlQuery);
+        cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("Treated Using Cryotherapy");
+
+        return cd;
+    }
+
+    /*Treated Using Thermocoagulation*/
+    public CohortDefinition treatedUsingThermocoagulationCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i where i.screening_result = 'Positive'" +
+                " and i.treatment_method = 'Thermocoagulation'  and visit_date between date(:startDate) and date(:endDate) group by i.patient_id;";
+
+        cd.setName("Thermocoagulation Treatment");
+        cd.setQuery(sqlQuery);
+        cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("Treated Using Thermocoagulation");
+
+        return cd;
+    }
+
+    /*Treated using LEEP*/
+    public CohortDefinition treatedUsingLEEPCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i where i.screening_result = 'Positive'" +
+                " and i.treatment_method = 'LEEP' and visit_date between date(:startDate) and date(:endDate) group by i.patient_id;";
+
+        cd.setName("LEEP Treatment");
+        cd.setQuery(sqlQuery);
+        cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("Treated using LEEP");
+
+        return cd;
+    }
+
+    /*Other Treatment Given*/
+    public CohortDefinition otherTreatmentGivenCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i where i.screening_result = 'Positive'" +
+                " and i.treatment_method = 'Other' and visit_date between date(:startDate) and date(:endDate) group by i.patient_id;";
+
+        cd.setName("Other Treatment");
+        cd.setQuery(sqlQuery);
+        cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("Other Treatment Given");
+
+        return cd;
+    }
+
+    /*HIV Positive Clients Screened*/
+    public CohortDefinition HIVPositiveClientsScreenedCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i inner join kenyaemr_etl.etl_hts_test t on t.patient_id = i.patient_id where t.final_test_result = 'Positive' and date(:startDate) and date(:endDate) group by i.patient_id;";
```

Comment:
1. There are patients who will not be in hts_test coz they are transfer-ins (known positive). Get these from hiv enrolment. So you need to left join the two secondary tables (through sub-queries) with screening table and in the where clause check for those who are in either hts or hv enrolment
2. The date part has a bug. Which date are you comparing with the date parameters?